### DISCUSSION TITLE
qrcode live stream in browser

### DISCUSSION THREAD

**Discussion by ghost** [ericblade/quagga2#482] [General]
Hi, there, interesting project! 

I was trying to detect qr codes at live-streamingi. It runs fine but with a few caveats that I detail below:

- Missing `decodePattern`. Adding `decodePattern(){ return null }` helps to silence it. Is there any recommended definition (i guess this would transform the code to text) ?

- The `readers` array complains in Typescript if I add any custom reader like `qrcode`, so I had to `@ts-expect-error` is there any way to do it better ?

The little  [ site to test ](https://santimirandarp.github.io/barcode_test/) and the short [source code here.](https://github.com/santimirandarp/barcode_test). To reproduce the first error it can just be cloned and remove `decodePattern`. 
Quagga2 stops at 1st detection (in purpose)

I'm ready to help if there is the need, but can't be sure I'll fix anything.

Have a good day.

**Comment by ericblade** [comment]
hmm... the recent changes to reader input didn't seem to trigger an error for me in the test repo.. we might need to alter that a little bit....

in quagga2-reader-qr, i  have a decodePattern() { return null } also, and it's commented with "needs to be implemented for Quagga in case it thinks there's a potential barcode in the image".  What was the error you were getting? 

I think if auto-detection was on, the default decodePattern would often treat a QR as some kind of a barcode, so it was messing with the results going into decodeImage.  Something like that.  It's been a couple of years since I looked at that part.
